docs/site/Express-middleware.md

```diff
@@ -0,0 +1,319 @@
+
+We create a few helper functions to wrap an Express middleware module into an
+LoopBack 4 interceptor.
+
+#### Adapt an Express middleware handler function to an interceptor
+
+```ts
+import {toInterceptor} from '@loopback/express';
+import morgan from 'morgan';
+
+const morganInterceptor = toInterceptor(morgan('combined'));
+```
+
+#### Create an interceptor from Express middleware factory function
+
+```ts
+import {createInterceptor} from '@loopback/express';
+import helmet, {IHelmetConfiguration} from 'helmet';
+
+const helmetConfig: IHelmetConfiguration = {};
+const helmetInterceptor = createInterceptor(helmet, helmetConfig);
+```
+
+If the Express middleware module does not expose a factory function conforming
+to the `ExpressMiddlewareFactory` signature, a wrapper can be created. For
+example:
+
+```ts
+import morgan from 'morgan';
+
+// Register `morgan` express middleware
+// Create a middleware factory wrapper for `morgan(format, options)`
+const morganFactory = (config?: morgan.Options) => morgan('combined', config);
+```
+
+### Define a provider class for middleware-based interceptor
+
+It's often desirable to allow dependency injection of middleware configuration
+for the middleware. We can use `defineInterceptorProvider` to simplify
+definition of such provider classes.
+
+```ts
+import {createInterceptor} from '@loopback/express';
+import helmet, {IHelmetConfiguration} from 'helmet';
+
+const helmetProviderClass = defineInterceptorProvider<IHelmetConfiguration>(
+  helmet,
+);
+```
+
+Alternatively, we can create a subclass of
+`ExpressMiddlewareInterceptorProvider`.
+
+```ts
+import {config} from '@loopback/context';
+import {
+  ExpressMiddlewareInterceptorProvider,
+  createMiddlewareInterceptorBinding,
+} from '@loopback/express';
+import helmet, {IHelmetConfiguration} from 'helmet';
+
+class HelmetInterceptorProvider extends ExpressMiddlewareInterceptorProvider<
+  IHelmetConfiguration
+> {
+  constructor(@config() helmetConfig?: IHelmetConfiguration) {
+    super(helmet, helmetConfig);
+  }
+}
+const binding = createMiddlewareInterceptorBinding(HelmetInterceptorProvider);
+app.add(binding);
+```
+
+#### Use Express middleware as invocation interceptors
+
+With the ability to wrap Express middleware as LoopBack 4 interceptors, we can
+use the same programming model to register middleware as global interceptors or
+local interceptors denoted by `@intercept` decorators at class and method
+levels.
+
+The middleware interceptor function can be directly referenced by `@intercept`.
+
+```ts
+import morgan from 'morgan';
+const morganInterceptor = toInterceptor(morgan('combined'));
+class MyController {
+  @intercept(morganInterceptor)
+  hello(msg: string) {
+    return `Hello, ${msg}`;
+  }
+}
+```
+
+It's also possible to bind the middleware to a context as a local or global
+interceptor.
+
+```ts
+import helmet, {IHelmetConfiguration} from 'helmet';
+const binding = registerExpressMiddlewareInterceptor(
+  app,
```

Comment:
Where is `app` coming from? Where should be this code snippet placed in a LB4 application?